src/clj/game/core/actions.clj

```diff
@@ -343,14 +343,17 @@
 (defn click-run
   "Click to start a run."
   [state side {:keys [server] :as args}]
-  (when (and (not (get-in @state [:runner :register :cannot-run]))
-             (can-run-server? state server)
-             (can-pay? state :runner "a run" :click 1))
-    (run state side server)
-    (let [cost (pay state :runner nil :click 1)
-          spent (build-spend-msg cost "make a run on")
-          message (str spent server)]
-      (system-msg state :runner message))))
+  (trigger-event state side :pre-click-run)
```

Comment:
Is it possible to avoid adding a new event by using turn events instead? Look at Service Outage for an example.